src/Dev/Tasks/TagsToShortcodeHelper.php

```diff
@@ -0,0 +1,290 @@
+            $this->validTagsPattern,
+            $this->validAttributesPattern
+        );
+        preg_match($pattern, $tag, $matches);
+        return $matches;
+    }
+
+    /**
+     * @param string $src
+     * @return null|ParsedFileID
+     * @throws \League\Flysystem\Exception
+     */
+    private function getParsedFileIDFromSrc($src)
+    {
+        $fileIDHelperResolutionStrategy = new FileIDHelperResolutionStrategy();
+        $fileIDHelperResolutionStrategy->setResolutionFileIDHelpers([
+            new HashFileIDHelper(),
+            new LegacyFileIDHelper(),
+            $defaultFileIDHelper = new NaturalFileIDHelper(),
+        ]);
+        $fileIDHelperResolutionStrategy->setDefaultFileIDHelper($defaultFileIDHelper);
+
+        // set fileID to the filepath relative to assets dir
+//        $pattern = '/^\\/?' . ASSETS_DIR . '?\\/?/';
+        $pattern = '/^\\' . DIRECTORY_SEPARATOR . '?' . ASSETS_DIR . '?\\' . DIRECTORY_SEPARATOR . '?/';
+        $fileID = preg_replace($pattern, '', $src);
+
+        // Try resolving with public filesystem first
+        $filesystem = $this->flysystemAssetStore->getPublicFilesystem();
+        $parsedFileId = $fileIDHelperResolutionStrategy->softResolveFileID($fileID, $filesystem);
+        if (!$parsedFileId) {
+            // Try resolving with protected filesystem
+            $filesystem = $this->flysystemAssetStore->getProtectedFilesystem();
+            $parsedFileId = $fileIDHelperResolutionStrategy->softResolveFileID($fileID, $filesystem);
+        }
+        return $parsedFileId;
+    }
+
+    /**
+     * @param string $tag
+     * @return string|null Returns the new tag or null if the tag does not need to be rewritten
+     */
+    private function getNewTag($tag)
+    {
+        $tuple = $this->getTagTuple($tag);
+        if (!isset($tuple['tagType']) || !isset($tuple['src'])) {
+            return null;
+        }
+        $tagType = $tuple['tagType'];
+        $src = $tuple['src'] ?: $tuple['href'];
+
+        // Search for a File object containing this filename
+        $parsedFileID = $this->getParsedFileIDFromSrc($src);
+        if (!$parsedFileID) {
+            return null;
+        }
+
+        /** @var File $file */
+        $file = File::get()->filter('FileFilename', $parsedFileID->getFilename())->first();
+        if (!$file) {
+            $file = Versioned::withVersionedMode(function () use ($parsedFileID) {
+                Versioned::set_stage(Versioned::LIVE);
+                return File::get()->filter('FileFilename', $parsedFileID->getFilename())->first();
+            });
+        }
+
+        if ($parsedFileID && $file) {
+            if ($tagType == 'img') {
+                $find = [
+                    '/(<|\[)img/',
+                    '/src\s*=\s*".*?"/',
+                    '/href\s*=\s*".*?"/',
+                    '/(>|\])/',
+                ];
+                $replace = [
+                    '[image',
```

Comment:
Images can have further attributes (e.g. a width/height)

Reply:
I've added unit tests for this use case.